team/bundles/org.eclipse.team.core/src/org/eclipse/team/internal/core/UserStringMappings.java

```diff
@@ -76,48 +85,67 @@ public void setStringMappings(String [] names, int [] types) {
 	public int getType(String string) {
 		if (string == null)
 			return Team.UNKNOWN;
-		final Integer type= referenceMap().get(string);
+		final Integer type = referenceMap().get(string);
 		return type != null ? type.intValue() : Team.UNKNOWN;
 	}
 
-	@Override
-	public void propertyChange(PropertyChangeEvent event) {
-		if(event.getProperty().equals(fKey))
-			fMap= null;
-	}
-
 	public void save() {
 		// Now set into preferences
 		final StringBuilder buffer = new StringBuilder();
 		final Iterator e = fMap.keySet().iterator();
 
 		while (e.hasNext()) {
-			final String filename = (String)e.next();
+			final String filename = (String) e.next();
 			buffer.append(filename);
 			buffer.append(PREF_TEAM_SEPARATOR);
 			final Integer type = fMap.get(filename);
 			buffer.append(type);
 			buffer.append(PREF_TEAM_SEPARATOR);
 		}
-		TeamPlugin.getPlugin().getPluginPreferences().setValue(fKey, buffer.toString());
+		try {
+			IEclipsePreferences node = InstanceScope.INSTANCE.getNode(TeamPlugin.ID);
+			node.put(fKey, buffer.toString());
+			node.flush();
+		} catch (BackingStoreException ex) {
+			TeamPlugin.log(IStatus.ERROR, ex.getMessage(), ex);
+		}
 	}
 
 	protected Map<String, Integer> loadMappingsFromPreferences() {
-		final Map<String, Integer> result= new HashMap<>();
+		final Map<String, Integer> result = new HashMap<>();
 
-		if (!fPreferences.contains(fKey))
+		if (!nodeAccessibleAndExists(fKey))
 			return result;
-
-		final String prefTypes = fPreferences.getString(fKey);
-		final StringTokenizer tok = new StringTokenizer(prefTypes, PREF_TEAM_SEPARATOR);
+		final StringTokenizer tok = new StringTokenizer(mappings(), PREF_TEAM_SEPARATOR);
 		try {
 			while (tok.hasMoreElements()) {
 				final String name = tok.nextToken();
-				final String mode= tok.nextToken();
+				final String mode = tok.nextToken();
 				result.put(name, Integer.valueOf(mode));
 			}
 		} catch (NoSuchElementException e) {
 		}
 		return result;
 	}
+
+	private String mappings() {
+		return Optional.ofNullable(InstanceScope.INSTANCE.getNode(TeamPlugin.ID)) //
+				.map(node -> node.get(fKey, null)) //
+				.orElse(EMPTY_PREF);
+	}
+
+	private boolean nodeAccessibleAndExists(String key) {
+		try {
+			IEclipsePreferences node = InstanceScope.INSTANCE.getNode(TeamPlugin.ID);
+			return Stream.of(node.keys()).anyMatch(candidate -> key.equals(candidate));
```

Comment:
why not use something like node.get(key,null)!=null ?

Reply:
Such a construction (like "any of the available keys matches the key we look for") looks more readable for me. But your suggestion is a good alternative too. Found nothing about it in contribution guidelines, but can change it to your way if it fits project code style better